.github/workflows/docs.yml

```diff
@@ -26,13 +36,77 @@ jobs:
+      - name: Set version env var to 'head'
+        if: github.event_name == 'push'
+        run: echo "DOCS_VERSION=head" >> $GITHUB_ENV
+
+      # If this was a release, set the version number
+      - name: Set version env var
+        if: github.event_name == 'release'
+        run: echo "DOCS_VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: Set version in _config.yml
+        run: |
+          echo "version: ${{ env.DOCS_VERSION }}" >> ./docs/_config.yml
+
+      # Only need to build the full list of versions every time there is a new release
+      - name: Update version.json list
+        if: github.event_name == 'release'
+        run: ls ./gh-pages | grep -E '^v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$' | sed 's/^/"/;s/$/"/' | tr '\n' ',' | sed 's/.$//' | sed 's/^/[/;s/$/]/' | jq '. += ["${{ env.DOCS_VERSION }}"] | sort_by(.) | reverse | . += ["latest", "head"]' > ./docs/assets/js/versions.json
+
+      - name: Install dependencies
+        working-directory: docs
+        run: bundle install
+
+      - name: Build doc site version
+        working-directory: docs
+        run: bundle exec jekyll build --baseurl /firefly/${{ env.DOCS_VERSION }}
+
+      - name: Check the docs for broken links (root)
+        if: github.event_name == 'pull_request'
+        working-directory: docs
+        run: bundle exec htmlproofer --allow-hash-href --assume-extension ./_site --url-swap '^/firefly/:/' --url-ignore /127.0.0.1/,/localhost/
+
+      - name: Check the docs for broken links (version)
+        if: github.event_name != 'pull_request'
+        working-directory: docs
+        run: bundle exec htmlproofer --allow-hash-href --assume-extension ./_site --url-swap '^/firefly/${{ env.DOCS_VERSION }}/:/' --url-ignore /127.0.0.1/,/localhost/
 
-      - name: Deploy
+      - name: Deploy docs (version)
         if: github.event_name == 'push'
         uses: peaceiris/actions-gh-pages@v3
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           publish_branch: gh-pages
-          publish_dir: ./docs/_site
\ No newline at end of file
+          publish_dir: ./docs/_site
+          destination_dir: ./${{ env.DOCS_VERSION }}
+
+      ## FINAL STEPS TO COMPLETE IF THIS WAS A NEW RELEASE ##
+
+      - name: Copy new version into version directory
+        if: github.event_name == 'release'
+        working-directory: docs
+        run: mv _site ${{ env.DOCS_VERSION }}
+
+      - name: Copy existing versions into working directory
+        if: github.event_name == 'release'
+        run: ls ./gh-pages | grep -E '^v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$' | xargs -I '{}' mv './gh-pages/{}' ./docs
```

Comment:
I see this is the same regex - does github pages have a variable construct we can use to have this in the code once?

Reply:
Yeah I can assign it to a variable. I'll make that change.